kafka-client/src/main/java/dev/responsive/kafka/internal/db/CassandraKeyValueTable.java

```diff
@@ -53,21 +53,27 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class CassandraKeyValueTable implements RemoteKVTable<BoundStatement> {
+public class CassandraKeyValueTable implements
+    RemoteKVTable<Integer, BoundStatement>,
+    RemoteLwtTable<Bytes, Integer, BoundStatement>{
```

Comment:
this kind of inheritance (when you inherit from two things that extend the same base interface) feels like a code smell to me. I suspect the reason you did this is so that you can reuse the `LwtWriter` - instead is there any reason why `RemoteLwtTable` needs to extend `RemoteTable`? if we make it its own interface then `LwtWriter` can accept both a `RemoteTable` and a `RemoteLwtTable` but we just pass in the same table once (if it implements both).I think then we can also rename the interface to something like `CassandraLwtCoordinator` to indicate that it's really specific to Cassandra's EOS implementation and better explain what it does (it isn't really a "table" after all)

Reply:
Yeah I agree this is probably the weakest part of the abstraction. I was actually wondering if it might make sense to have the Mongo EOS implementation share the `RemoteLwtTable` interface as well, or if we should fully commit to this as C*/Lwt specific

Let's chat about this when we go over it

Reply:
Changed the `RemoteLwtTable` to a standalone interface (that doesn't extend `RemoteTable` itself), and also renamed it to `TableMetadata` since it is currently being used to pass along the stream-time in addition to the epoch

I'm not crazy about the new name and think we might be able to clean this up further, but as discussed, I'm going to revisit this part during the followup PR to persist the stream-time